atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/impl/SnapshotTransaction.java

```diff
@@ -1114,24 +1114,27 @@ private int estimateSize(List<RowResult<Value>> rangeRows) {
                 } else {
                     // The value has a commit timestamp less than our start timestamp, and is visible and valid.
                     if (value.getContents().length != 0) {
-                        found++;
                         results.put(key, transformer.apply(value));
+                        keysAddedBuilder.add(key);
                     }
                 }
             }
         }
-        count.set(found);
+        Set<Cell> keysAddedToResults = keysAddedBuilder.build();
+        count.set(keysAddedToResults.size());
```

Comment:
This will change the size of `count` across iterations, because we're modifying `rawResults` across calls.
I think `count.accumulateAndGet(c -> c + keysAddedToResults.size())` should work.